packages/atproto_client/models/string_formats.py

```diff
@@ -193,13 +206,7 @@ def validate_nsid(v: str, _: ValidationInfo) -> str:
     Raises:
         ValueError: If NSID format is invalid
     """
-    if (
-        not atproto_core_validate_nsid(v, soft_fail=True)
-        or len(v) > MAX_NSID_LENGTH
-        or any(c in v for c in '@_*#!')  # Explicitly disallow special chars
-        or any(len(seg) > 63 for seg in v.split('.'))  # Max segment length
-        or any(seg[-1].isdigit() for seg in v.split('.'))  # No segments ending in numbers
-    ):
+    if not atproto_core_validate_nsid(v, soft_fail=True):
```

Comment:
i was causing failures by incorrectly reapplying bits of logic that now live in the core validation